CMakeLists.txt

```diff
@@ -32,64 +35,38 @@ set(FETCHCONTENT_BASE_DIR ${CMAKE_CURRENT_LIST_DIR}/lib/_deps)
 set(FETCHCONTENT_QUIET OFF)
 set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
 set(BUILD_SHARED_LIBS OFF)
-set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(BUILD_TESTING OFF)
 
-# Make bayesmix-dev/math available
-message(CHECK_START "Fetching math")
-FetchContent_Declare(math
-  GIT_REPOSITORY "https://github.com/bayesmix-dev/math.git"
-  GIT_TAG "develop"
-)
-FetchContent_MakeAvailable(math)
-message(CHECK_PASS "fetched")
-
-# TBB CMake integration
-message(STATUS "Using math TBB")
-# Define TBB_ROOT Folder
-set(TBB_ROOT ${math_SOURCE_DIR}/lib/tbb)
-file(COPY ${math_SOURCE_DIR}/lib/tbb_2020.3/ DESTINATION ${TBB_ROOT})
-# Build TBB Library with CMake Integration
-include(${TBB_ROOT}/cmake/TBBBuild.cmake)
-list(APPEND MAKE_ARGS "tbb_build_dir=${TBB_ROOT}")
-list(APPEND MAKE_ARGS "tbb_build_prefix=tbb")
-tbb_build(TBB_ROOT ${TBB_ROOT} CONFIG_DIR TBB_DIR MAKE_ARGS ${MAKE_ARGS})
-# Require TBB library (for compile / link options)
-find_package(TBB)
-
-# Check if Protobuf is present in system
-find_package(Protobuf)
-if (NOT Protobuf_FOUND AND NOT PROTOBUF_FOUND AND NOT TARGET protobuf::libprotobuf)
-  
-  # if (${CMAKE_VERSION} VERSION_LESS "3.20.0")
-  #   message(FATAL_ERROR
-  #           "Your cmake version is too old: either install a newer version (>=3.20)"
-  #           " or install google's protocol buffer (protobuf) library by hand.")
-  # endif()
-
-  message(CHECK_START "Fetching Protobuf")
-  list(APPEND CMAKE_MESSAGE_INDENT "  ")
-
-  option(protobuf_BUILD_TESTS "" OFF)
-  set(protobuf_BUILD_EXPORT OFF)
-  set(protobuf_MSVC_STATIC_RUNTIME OFF)
-  FetchContent_Declare(protobuf
-    GIT_REPOSITORY "https://github.com/protocolbuffers/protobuf.git"
-    GIT_TAG "v3.14.0"
-    GIT_SUBMODULES ""
-    SOURCE_SUBDIR cmake)
-  FetchContent_MakeAvailable(protobuf)
-
-  list(POP_BACK CMAKE_MESSAGE_INDENT)
-  message(CHECK_PASS "fetched")
-  message("Protobuf_FOUND ${Protobuf_FOUND}")
-  message("   --> PROTOBUF LIB: ${PROTOBUF_LIBRARIES}")
-  message("   --> PROTOBUF INCLUDE: ${PROTOBUF_INCLUDE_DIRS}")
-  message("   --> PROTOBUF VERSION: ${Protobuf_VERSION}")
-  message("   --> PROTOBUF Found: ${Protobuf_FOUND}")
+# Set BASEPATH variable
+set(BASEPATH "${CMAKE_CURRENT_LIST_DIR}")
+
+# Make available bayesmix-dev/math (and TBB)
+include(cmake/math.cmake)
```

Comment:
just being picky but I think it makes more sense to have another PR dedicated to the building process.